lib/zlib.js

```diff
@@ -173,14 +175,13 @@ function zlibOnError(message, errno, code) {
   const self = this[owner_symbol];
   // There is no way to cleanly recover.
   // Continuing only obscures problems.
-  _close(self);
-  self._hadError = true;
 
   // eslint-disable-next-line no-restricted-syntax
   const error = new Error(message);
   error.errno = errno;
   error.code = code;
-  self.emit('error', error);
+  self.destroy(error);
+  self[kError] = error;
```

Comment:
What tests would we break if we did not do this?

Reply:
<details>
```
if [ ! -r node -o ! -L node ]; then ln -fs out/Release/node node; fi
=== release test-zlib-from-gzip-with-trailing-garbage ===                     
Path: parallel/test-zlib-from-gzip-with-trailing-garbage
out/Release/node[11889]: ../../src/node_zlib.cc:339:void node::(anonymous namespace)::CompressionStream<node::(anonymous namespace)::ZlibContext>::Write(uint32_t, char *, uint32_t, char *, uint32_t) [CompressionContext = node::(anonymous namespace)::ZlibContext, async = false]: Assertion `!closed_ && "already finalized"' failed.
 1: 0x1011c9625 node::Abort() (.cold.1) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 2: 0x1000cad79 node::Abort() [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 3: 0x1000cabe1 node::Assert(node::AssertionInfo const&) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 4: 0x10015d7f4 void node::(anonymous namespace)::CompressionStream<node::(anonymous namespace)::ZlibContext>::Write<false>(v8::FunctionCallbackInfo<v8::Value> const&) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 5: 0x100294858 v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 6: 0x100293dec v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 7: 0x100293512 v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 8: 0x1009e49b9 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
Command: out/Release/node /Users/ronagy/GitHub/nxtedition/node/test/parallel/test-zlib-from-gzip-with-trailing-garbage.js
--- CRASHED (Signal: 6) ---
=== release test-zlib-truncated ===                               
Path: parallel/test-zlib-truncated
out/Release/node[11900]: ../../src/node_zlib.cc:339:void node::(anonymous namespace)::CompressionStream<node::(anonymous namespace)::ZlibContext>::Write(uint32_t, char *, uint32_t, char *, uint32_t) [CompressionContext = node::(anonymous namespace)::ZlibContext, async = false]: Assertion `!closed_ && "already finalized"' failed.
 1: 0x1011c9625 node::Abort() (.cold.1) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 2: 0x1000cad79 node::Abort() [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 3: 0x1000cabe1 node::Assert(node::AssertionInfo const&) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 4: 0x10015d7f4 void node::(anonymous namespace)::CompressionStream<node::(anonymous namespace)::ZlibContext>::Write<false>(v8::FunctionCallbackInfo<v8::Value> const&) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 5: 0x100294858 v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 6: 0x100293dec v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 7: 0x100293512 v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
 8: 0x1009e49b9 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit [/Users/ronagy/GitHub/nxtedition/node/out/Release/node]
Command: out/Release/node /Users/ronagy/GitHub/nxtedition/node/test/parallel/test-zlib-truncated.js
--- CRASHED (Signal: 6) ---
[00:01|% 100|+  45|-   2]: Done 
```  
</details>